docs/product-repo-plan-proposal.md

```diff
@@ -0,0 +1,56 @@
+- Setup time for adding performance testing is as simple as adding a new job that references the yaml template that we provide
+
+## Procedural
+
+- Combine the yaml we already use and, likely, the yaml that has been worked on in coreclr (why reinvent the wheel), with any additional logic necessary for pulling down and running the performance repository in product repos
+  - yaml will need to include any additional parameters/knobs that the product repos may want to turn (ie tiered/non-tiered/minopts, pgo/no-pgo, etc)
+  - yaml will need to include all uploading to benchview logic
+  - yaml should be as barebones as possbile with most of the work done in the proj file
+  - yaml should allow sending of just the performance repo as well as sending the performance repo and the core_root directory (for repos that need to specify that)
+- Add a send-perf-to-helix.proj file that will run the benchmarks in a sane way
+  - proj file should include most of the running logic
+  - proj file should allow for both partitioned runs and non-partitioned runs (there's no real reason for functional testing to run partitioned since they run so quickly, but anything where we are collecting actual performance data should be partitioned)
+  - proj file should accept any csproj from the performance repo so that when new benchmark categories are added (ie ml, scenarios, etc), it is simple to start running those new benchmarks
+- Add these two templates (the yaml and project) to arcade
+- Let the product repo owners, or all of .NET Core Engineering Partners, that the yaml is ready for their use
+
+## Pros
+
+- We control what they run
+- It's a clean solution
+- It uses arcade the way it is meant to be used (sharing shared infrastructure)
+- Product repos can run per-commit and on PR without having to go outside of their repo
+- The performance repo does not have to know about the product repos
+- We can use scripting that we already have, modify it to allow a supplied corerun, and almost immediately empower the product repos to start performance testing
+- We don't have to know anything about the builds of the product repos
+- When new repos want to start running performance testing, we don't have to do anything except point them to the template
+- We do not have to work around darc/dependency flow
+- We do not need to know why the repo triggered a job (ie because coreclr/corefx/etc changed) because the triggers will be in coreclr/corefx
+- While we are still on benchview, this obfuscates a lot of the upload logic so that product repos don't have to worry about getting it wrong
+- When we move off Benchview, this will allow us to have a seemless experience for product repos
+
+## Cons
+
+- This still could enable the product repos to do their own thing for perf testing (ie by not using our template)
+- Scripting might not be quite general enough to work for anyone
+- The product repos will have to pull down the performance repo to do work
```

Comment:
is this a point in time problem? Once we've moved to a place where we don't have to build benchmark.net on the end machine, could we build the repo on the build machine and just push down artifacts (like functional tests?)

Reply:
We'd still need to sync the perf repo into the build machine of course.

Reply:
Yes, I meant your second comment with this point. And sort of like - we'll need a way to know where in the perf repo we are. I mentioned to Drew adding an additional piece of metadata to the run info that is the perf repo hash, which we could use to easily know if the perf benchmarks moved underneath us between two runs (but not actually break graphs or anything). He thinks that should be no problem